keyserver/listener.go

```diff
@@ -0,0 +1,127 @@
+package keyserver
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net"
+	"time"
+
+	. "github.com/coniks-sys/coniks-go/protocol"
+)
+
+func (server *ConiksServer) listenForRequests(ln net.Listener, handler func(msg []byte) ([]byte, error)) {
```

Comment:
I still think that `<-server.stop` should be handled in a go routine. And your suggestion on a second wait group for the accepted requests seems a safe/good way to go.